### PR TITLE
worker: Add skeleton service

### PR DESCRIPTION
**Pull request by efritz**
This PR adds a (currently empty) service that will eventually be the home to all "miscellaneous background tasks".

Fixes https://github.com/sourcegraph/sourcegraph/issues/21757.

### CODE REVIEW

**Review by tsenart — APPROVED** [review]

File: cmd/worker/shared/main.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/sourcegraph/sourcegraph/internal/goroutine"
+	"github.com/sourcegraph/sourcegraph/internal/httpserver"
+	"github.com/sourcegraph/sourcegraph/internal/logging"
+	"github.com/sourcegraph/sourcegraph/internal/trace"
+	"github.com/sourcegraph/sourcegraph/internal/tracer"
+)
+
+const addr = ":3189"
+
+func Main() {
```
Comment (ggilmore) [inline cmd/worker/shared/main.go:19]: Thoughts on using Main() as a name instead of something like Start()?

Comment (efritz) [reply cmd/worker/shared/main.go:19]: That would be fine, I was just copying the frontend setup to get started.

Comment (efritz) [reply cmd/worker/shared/main.go:19]: Renamed in a follow-up PR.

File: cmd/worker/shared/main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	env.HandleHelpFlag()
+	logging.Init()
+	tracer.Init()
+	trace.Init(true)
+
+	// Start debug server
+	ready := make(chan struct{})
+	go debugserver.NewServerRoutine(ready).Start()
+
+	var allRoutines []goroutine.BackgroundRoutine
```
Comment (ggilmore) [inline cmd/worker/shared/main.go:30]: Interesting. Is there going to be a concept of registering routines after this service has started? Or, is it all static (known ahead of time)? Does this imply an implicit ordering on services?

Comment (efritz) [reply cmd/worker/shared/main.go:30]: All static routines that can be enabled or disabled via the environment. If you need to make tasks at runtime you can call your own `MonitorBackgroundRoutine` after constructing the list. So far we haven't needed trees of things and I don't see it being an immediate need.

Initialization and ordering isn't super well-defined yet. I suspect that there will be some shared setup that needs to happen, and the easiest (lowest-effort) way to do that is to have some initialization routine (setup db connections, etc) guarded by a `sync.Once`.

Comment (efritz) [reply cmd/worker/shared/main.go:30]: Some more concrete context in https://github.com/sourcegraph/sourcegraph/pull/21774.

**Pull request merged**